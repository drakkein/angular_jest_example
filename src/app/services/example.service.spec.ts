import { TestBed, async } from "@angular/core/testing";

import { ExampleService } from './example.service';

import { first } from 'rxjs/operators';

describe('ExampleService', () => {
    let service: ExampleService;
    beforeEach(async(() => {
        TestBed.configureTestingModule({
            providers: [
                ExampleService,
            ],
        }).compileComponents();
    }));

    beforeEach(() => {
        service = TestBed.get(ExampleService);
    });

    it('Should match snapshot', () => {
        const data = service.getList()
            .pipe(first())
            .subscribe(v => {
                expect(v).toMatchSnapshot();
            })
    });
});