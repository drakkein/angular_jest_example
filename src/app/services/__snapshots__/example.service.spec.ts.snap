// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleService Should match snapshot 1`] = `
Array [
  Object {
    "foo": "bar",
  },
  Object {
    "foo": "lorem",
  },
  Object {
    "foo": "test",
  },
]
`;
